mono/sgen/sgen-array-list.h

```diff
@@ -49,14 +49,31 @@ typedef struct {
 	int mem_type; /* sgen internal mem type or -1 for malloc allocation */
 } SgenArrayList;
 
+#if !defined(ENABLE_MSVC_LZCNT) && defined(_MSC_VER)
+static inline guint32
+sgen_msvc_lzcnt (guint32 x)
+{
+	gulong leading_zero_bits;
+	return _BitScanReverse (&leading_zero_bits, (gulong)x) ? 31 - leading_zero_bits : 32;
+}
+#define __builtin_clz sgen_msvc_lzcnt
+#elif defined(ENABLE_MSVC_LZCNT) && defined(_MSC_VER)
+static inline guint32
+sgen_msvc_lzcnt (guint32 x)
+{
+	return __lzcnt (x);
+}
+#define __builtin_clz sgen_msvc_lzcnt
```

Comment:
Doesn't feel like a good idea to define __builtin_clz. We should use sgen_msvc_lzcnt in sgen, and have it defined to __builtin_clz or whatever intrinsics are present on other platforms. Maybe also have naive implementation for it on unknown platform and avoid define in sgen_array_list_index_bucket.

Reply:
Sure, I can give it a different name and move it so it could be shared within sgen, put it in sgen-gc.h?

Reply:
It can also stay in sgen-array-list.h if not used anywhere else. I just don't want to redefine __builtin_clz which is meant to be a compiler intrinsic.

Reply:
OK, then I will keep it in sgen-array-list.h and just it a different name.